addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -0,0 +1,134 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Progressing",type=string,JSONPath=`.status.conditions[?(@.type=="Progressing")].status`
+// +kubebuilder:printcolumn:name="Available",type=string,JSONPath=`.status.conditions[?(@.type=="Available")].status`
+
+// ManagedClusterAddOn is the Custom Resource object which holds the current state
+// of an add-on. This object is used by add-on operators to convey their state.
+// The resource should be created in the ManagedCluster namespace.
+type ManagedClusterAddOn struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec holds configuration that could apply to any operator.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterAddOnSpec `json:"spec"`
+
+	// status holds the information about the state of an operator.  It is consistent with status information across
+	// the Kubernetes ecosystem.
+	// +optional
+	Status ManagedClusterAddOnStatus `json:"status"`
+}
+
+// ManagedClusterAddOnSpec is empty for now.
+type ManagedClusterAddOnSpec struct {
+}
```

Comment:
I expected to see the CRD reference information and the display name and description also listed here.  The idea being that they fields are managed at the cluster scope, but this resource provides one-stop-shopping for "what does this mean".

Reply:
this information exists in the ClusterManagementAddon resource which have a tight relationship with the ManageClusterAddon resource, do we need to duplicate the information here? or can we link to the ClusterManagementAddon resource?

Reply:
the information is captured in the ClusterManagementAddon would it be sufficient to link out to that resource or should we duplicate the information here

Reply:
The `ManagedClusterAddOn` is a namespace-scoped object that will be created per cluster. I would expect that it has some awareness of its cluster-scoped ClusterManagementAddOn component (where the CRD) reference is defined, but I would expect here we would have a reference to the CR instance of the actual CRD. 

e.g. 

`ClusterManagementAddOn` --> Here's a new cluster management subsystem (cluster scoped) & I look for CRDs named "CRDNAME".
`ManagedClusterAddOn` --> I'm part of cluster management subsystem CLUSTERMGMTADDON_NAME and in particular, I'm here to let you know that you can configure my ClusterManagementAddOn for CLUSTER_NAME with this instance: CRDNAME_INSTANCE (namespace scoped).

Reply:
the linkage between ManagedClusterAddOn and ClusterManagmentAddOn is by name the 2 resource will have the same name

Reply:
the CR instance of the actual CRD will be referenced in the relatedObjects field in the status as we dont expect this to be a user configured parameter

Reply:
If a user creates a `ManagedClusterAddOn` and a `ClusterManagmentAddOn` of the same name isn't found, will a validating webhook reject the creation? Having them associated by name alone may be hard for users to predict. By convention, using the same name is fine -- but as an artifact to build a relationship in a model, it could be errorproned.